src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -0,0 +1,504 @@
+            and jnp.all(self.K >= 0.0)
+            and jnp.all(self.D >= 0.0)
+        )
+
+
+@jax_dataclasses.pytree_dataclass
+class RigidContactsState(ContactsState):
+    """Class storing the state of the rigid contacts model."""
+
+    def __eq__(self, other: RigidContactsState) -> bool:
+        return hash(self) == hash(other)
+
+    @staticmethod
+    def build(**kwargs) -> RigidContactsState:
+        """Create a `RigidContactsState` instance"""
+
+        return RigidContactsState()
+
+    @staticmethod
+    def zero(**kwargs) -> RigidContactsState:
+        """Build a zero `RigidContactsState` instance from a `JaxSimModel`."""
+        return RigidContactsState.build()
+
+    def valid(self, **kwargs) -> bool:
+        return True
+
+
+@jax_dataclasses.pytree_dataclass
+class RigidContacts(ContactModel):
+    """Rigid contacts model."""
+
+    parameters: RigidContactParams = dataclasses.field(
+        default_factory=RigidContactParams
+    )
+
+    terrain: jax_dataclasses.Static[Terrain] = dataclasses.field(
+        default_factory=FlatTerrain
+    )
+
+    @staticmethod
+    def detect_contacts(
+        W_o_C: jtp.Array,
+        W_o_dot_C: jtp.Array,
+        terrain_height: jtp.Array,
+        terrain_normal: jtp.Array,
+    ) -> tuple[jtp.Vector, tuple[Any, ...]]:
+        """
+        Detect contacts between the collidable points and the terrain.
+
+        Args:
+            W_o_C: The position of the collidable points.
+            W_o_dot_C: The linear velocity of the collidable points.
+            terrain_height: The height of the terrain at the collidable point position.
+            terrain_normal: The normal of the terrain at the collidable point position.
+
+        Returns:
+            A tuple containing the activation state of the collidable points along contact normal penetration depth and velocity.
+        """
+
+        # TODO: reduce code duplication with js.contact.in_contact
+        def detect_contact(
+            W_o_C: jtp.Array,
+            W_o_dot_C: jtp.Array,
+            terrain_height: jtp.Float,
+            terrain_normal: jtp.Vector,
+        ) -> tuple[jtp.Vector, tuple[Any, ...]]:
+            """
+            Detect contacts between the collidable points and the terrain."""
+
+            # Unpack the position of the collidable point.
+            _px, _py, pz = W_o_C.squeeze()
+            W_o_dot_C = W_o_dot_C.squeeze()
+
+            inactive = pz > terrain_height
+
+            # Compute the terrain normal and the contact depth
+            n̂ = terrain_normal.squeeze()
+            h = jnp.array([0, 0, terrain_height - pz])
+
+            # Compute the penetration depth normal to the terrain.
+            δ = jnp.maximum(0.0, jnp.dot(h, n̂))
+
+            # Compute the penetration normal velocity.
+            δ̇ = -jnp.dot(W_o_dot_C, n̂)
+
+            return inactive, (δ, δ̇)
+
+        inactive_collidable_points, (delta, delta_dot) = jax.vmap(detect_contact)(
```

Comment:
Feel free to use the delta dot symbol also here.

Reply:
Done in e1e0c6c9eea88e9d13b395a0a1e79bc73f620c0e